Content.Server/Info/RulesManager.cs

```diff
@@ -1,15 +1,55 @@
-﻿using Content.Shared.Info;
-using Robust.Shared.IoC;
+﻿using System.Net;
+using Content.Server.Database;
+using Content.Shared.Info;
 using Robust.Shared.Network;
 
 namespace Content.Server.Info;
 
 public sealed class RulesManager : SharedRulesManager
 {
+    [Dependency] private readonly IServerDbManager _dbManager = default!;
     [Dependency] private readonly INetManager _netManager = default!;
 
+    private static DateTime LastValidReadTime => DateTime.UtcNow - TimeSpan.FromDays(60);
+
+    private readonly Dictionary<NetUserId, DateTime> _lastReadRulesCache = new();
+
     public void Initialize()
     {
+        _netManager.RegisterNetMessage<ShouldShowRulesPopupMessage>();
         _netManager.RegisterNetMessage<ShowRulesPopupMessage>();
+        _netManager.RegisterNetMessage<RulesAcceptedMessage>(OnRulesAccepted);
+        _netManager.Connected += OnConnected;
+    }
+
+    private async void OnConnected(object? sender, NetChannelArgs e)
+    {
+        if (IPAddress.IsLoopback(e.Channel.RemoteEndPoint.Address))
+        {
+            return;
+        }
```

Comment:
this seems like a strange way to check for it, feels like it should check for admin or something

Reply:
We discussed this in Discord. You're not guaranteed to be an admin on a local machine, if you're testing something. Showing it even once would be annoying so this feels more fool proof.